timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -136,19 +140,21 @@ def static_features(self, value: Optional[pd.DataFrame]):
 
     @property
     def freq(self):
-        ts_index = self.index.levels[1]  # noqa
-        freq = (
-            ts_index.freq
-            or ts_index.inferred_freq
-            or self.loc[0].index.freq  # fall back to freq of first item
-            or self.loc[0].index.inferred_freq
-        )
-        if freq is None:
-            raise ValueError("Frequency not provided and cannot be inferred")
-        if isinstance(freq, str):
-            return freq
-        elif isinstance(freq, pd._libs.tslibs.BaseOffset):
-            return freq.freqstr
+        if self._cached_freq:
+            return self._cached_freq
```

Comment:
Do I understand correctly that if we have an irregularly-sampled TS, we will set `self._cached_freq = None` and therefore rerun lines 143-157 each time we access `df.freq`?

If this is the case, should we use some other token to denote irregularly-sampled TS (like `"None"` or `nan`), or is the overhead negligible?